src/sensitivity_interface.jl

```diff
@@ -67,17 +69,90 @@ instead of just at discrete data points.
       saving is required, one should solve dense `solve(prob,alg)`, use the
       solution in the adjoint, and then `sol(ts)` interpolate.
 
-### Syntax
+## Mathematical Definition
 
-For discrete adjoints, use:
+Adjoint sensitivity analysis finds the gradient of a cost function ``G`` defined by the
+infinitesimal cost over the whole time period, given by the equation:
+
+```math
+G(u,p)=G(u(\cdot,p))=\int_{t_{0}}^{T}g(u(t,p),p,t)dt
+```
+
+It does so by solving the adjoint problem:
+
+```math
+\frac{d\lambda^{\star}}{dt}=g_{u}(u(t,p),p)-\lambda^{\star}(t)f_{u}(t,u(t,p),p),\thinspace\thinspace\thinspace\lambda^{\star}(T)=0
+```
+
+and obtaining the sensitivities through the integral:
+
+```math
+\frac{dG}{dp}=\int_{t_{0}}^{T}\lambda^{\star}(t)f_{p}(t)+g_{p}(t)dt+\lambda^{\star}(t_{0})u_{p}(t_{0})
+```
+
+As defined, that cost function only has non-zero values over nontrivial intervals. However, in many
+cases one may want to include in the cost function loss values at discrete points, for example, matching
+the data at time points `t`. In this case, terms of `g` can be represented by Dirac delta functions
+which are then applied in the corresponding ``\lambda^\star`` and ``\frac{dG}{dp}`` equations. 
+
+For more information, see [Sensitivity Math Details](@ref sensitivity_math).
+
+## Positional Arguments
+
+- `sol`: the solution from the forward pass of the ODE. Note that if not using a checkpointing
+  sensitivity algorithm, then it's assumed that the (dense) interpolation of the forward solution is
+  of sufficient accuracy for recreating the solution at any time point.
+- `alg`: the algorithm to use for the solution of the adjoint problem.
+
+## Keyword Arguments
+
+- `t`: the time points at which the discrete cost function is to be evaluated.
+  This argument is only required if discrete cost functions are declared.
+- `g`: the continuous instantanious cost ``g(u,p,t)`` at a given time point
+  represented by a Julia function `g(u,p,t)`. This argument is only required
+  if there is a continuous 
+- `dgdu_discrete`: the partial derivative ``g_u`` evaluated at the discrete
+  (Dirac delta) times. If discrete cost values are given, then `dgdu_discrete`
+  is required.
+- `dgdp_discrete`: the partial derivative ``g_p`` evaluated at the discrete
+  (Dirac delta) times. If discrete cost values are given, then `dgdp_discrete`
+  is not required ans is assumed to be zero.
+- `dgdu_continuous`: the partial derivative ``g_u`` evaluated at times
+  not corresponding to terms with an associated Dirac delta. If `g` is given,
+  then this term is not required and will be approximated by numerical or autoamatic
```

Comment:
```suggestion
  then this term is not required and will be approximated by numerical or (forward-mode) automatic
```

Reply:
This is maybe a tiny thing we can improve --i.e., adding the option that `gradient!` calls can also be done by reverse-mode AD 
https://github.com/SciML/SciMLSensitivity.jl/blob/14fb96ea03ef70b224a5cb7a63b09c04ac82ebd6/src/derivative_wrappers.jl#L164

Reply:
Indeed that would be an improvement. Worth adding an issue on. I don't think that's the biggest improvement though since the g function is usually "sufficiently simple", but it's good to tie a bow on it.

Reply:
That might be good to label as good for a GSoC project as a starter issue but isn't something that's on the "main path" of the development, so if we don't get to it it's fine but it's also not too hard to solve